compiler/rustc_mir/src/borrow_check/diagnostics/move_errors.rs

```diff
@@ -492,8 +492,8 @@ impl<'a, 'tcx> MirBorrowckCtxt<'a, 'tcx> {
             {
                 if let Ok(pat_snippet) = self.infcx.tcx.sess.source_map().span_to_snippet(pat_span)
                 {
-                    if pat_snippet.starts_with('&') {
-                        let pat_snippet = pat_snippet[1..].trim_start();
+                    if let Some(stripped) = pat_snippet.strip_prefix('&') {
+                        let pat_snippet = stripped.trim_start();
                         let (suggestion, to_remove) = if pat_snippet.starts_with("mut")
                             && pat_snippet["mut".len()..].starts_with(rustc_lexer::is_whitespace)
```

Comment:
this case is missed

Reply:
The lint did not report this because 
````rust
let (suggestion, to_remove) = if let Some(stripped) = pat_snippet.strip_prefix("mut")
                            && stripped.starts_with(rustc_lexer::is_whitespace)
                        {
                            (stripped.trim_start(), "&mut")
                        } else {
                            (pat_snippet, "&")
                        };
````
would require unstable features like `let_chains`

Reply:
You could use `strip_prefix(..).map`. However, it doesn't need to change anything in this PR.
Just a suggestion for clippy.